src/auth/redisClient.ts

```diff
@@ -0,0 +1,80 @@
+const redis = require("redis");
+const crypto = require("crypto");
+const bluebird = require("bluebird");
+const cryptojs = require("crypto-js");
+
+const HOUR_IN_SECONDS = 3600;
+
+// Adding promises to redis
+// https://stackoverflow.com/a/54844935
+
+// https://cloud.google.com/community/tutorials/nodejs-redis-on-appengine
+const client = (() => {
+  bluebird.promisifyAll(redis);
+  if (process.env.REDIS_PORT) {
+    const authParams: {
+      [index: string]: any;
+    } = {};
+    if (process.env.REDIS_KEY) {
+      authParams.auth_pass = process.env.REDIS_KEY;
+      authParams.return_buffers = true;
+    }
+    return redis.createClient(
+      process.env.REDIS_PORT,
+      process.env.REDIS_HOST,
+      authParams
+    );
+  } else {
+    return redis.createClient();
+  }
+})();
+
+const hash = (text: any) =>
+  crypto.createHash("sha256").update(text).digest("hex");
+const defaultKeyWord = "12345";
```

Comment:
Let's force the ENV to exist. We don't want to accidentally forget it and use 12345 as the key.